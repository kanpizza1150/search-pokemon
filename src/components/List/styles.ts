import styled from 'styled-components'
import {
  setFlex,
  setTransition,
  setColor,
  borderStyle,
} from '../../utils/globalStyles'

export const ListWrapper = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: repeat(auto-fill, 40rem);
  grid-gap: 0.1rem 2rem;
  justify-content: center;
  margin: 0 auto;
  ${setTransition()};
`
export const ModalWrapper = styled.div`
  background-color: ${setColor.white};
  height: 60rem;
  width: 100%;
  ${setFlex()};
  flex-direction: column;
  ${borderStyle};
  .word-not-found {
    font-size: 3rem;
    color: ${setColor.grey};
    span {
      color: ${setColor.black};
      font-weight: bold;
    }
  }
`
