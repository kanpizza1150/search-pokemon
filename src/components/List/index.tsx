import React from 'react'
import { ListWrapper } from './styles'
import Card from '../Card'
import Modal from '../Modal'
import { IPokemon } from '../../utils/interfaces'

interface IProps {
  pokemons: []
  searchKey: string
}

const List: React.FC<IProps> = ({ pokemons = [], searchKey }) => {
  const _renderCard = () => {
    let result: JSX.Element[] | JSX.Element
    if (pokemons.length > 0) {
      result = pokemons.map((pokemon: IPokemon) => (
        <Card key={pokemon.id} pokemon={pokemon} />
      ))
    } else {
      result = (
        <Modal>
          <div className='word-not-found'>
            <span>{searchKey}</span> is not found ‚ùå
          </div>
        </Modal>
      )
    }
    return result
  }

  return <ListWrapper>{_renderCard()}</ListWrapper>
}
export default List
