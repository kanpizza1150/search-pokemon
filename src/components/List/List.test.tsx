import { fireEvent, render, waitForElement } from '@testing-library/react'
import List from './index'

let mockedUseQuery = {
  loading: false,
  data: {},
  error: {},
}
jest.mock('@apollo/client', () => ({
  useQuery: () => mockedUseQuery,
  gql: jest.fn(),
  ApolloProvider: jest.fn((props) => (
    <apolloprovider>{props.children}</apolloprovider>
  )),
}))

jest.mock('../Card', () => ({ children }) => (
  <card data-testid='card'>{children}</card>
))

let mockedPokemons = []

describe('List Component', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<List pokemons={mockedPokemons} />)
    expect(asFragment).toMatchSnapshot()
  })
  it('Should render not found element', async () => {
    const { getByTestId } = render(<List pokemons={mockedPokemons} />)
    expect(getByTestId('notfound').textContent).toMatch(/is not found/)
  })
  it('Should render cards section', () => {
    mockedPokemons = [{}]
    const { getByTestId } = render(<List pokemons={mockedPokemons} />)
    expect(getByTestId('cards')).toBeInTheDocument()
  })
  it('Should render 10 cards', async () => {
    mockedPokemons = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
    const { getAllByTestId } = render(
      <List pokemons={mockedPokemons} limit={10} />
    )
    expect(getAllByTestId('card').length).toBe(10)
  })
})
