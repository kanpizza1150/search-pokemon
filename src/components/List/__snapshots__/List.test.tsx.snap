// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Should match snapshot 1`] = `[Function]`;
